### PR TITLE
Some smaller changes.

### PR DESCRIPTION
**Pull request by jochumdev**
3 patches i wrote for you great network formula:
- a way to add/remove packages trough network.packages
- some more known interface values.
- add a way to edit dhclient.conf via network.interfaces

### CODE REVIEW

File: pillar.example.sls [hunk 1]
```diff
@@ -81,3 +81,10 @@ network:
           netmask: 255.255.0.0
           gateway: 192.168.2.1
 
+  # To add/remove packages here.
+  # You need to include the state network.packages in your salt.
+  pkgs:
```
Comment (arnisoph) [inline pillar.example.sls:86]: Can we achieve this with https://github.com/bechtoldt/saltstack-tools-formula?

**Comment by pcdummy** [comment]
Hi, yeah salstack-tools could make the same as network.packages, but i like it to have everything at the right place what do you think about it?

**Comment by bechtoldt** [comment]
Normally I would say preventing code duplication is a good thing. Do we have technical reasons to depend on pkg mgmt in the network formula?

**Comment by pcdummy** [comment]
This way i know that these packages will be installed/removed because of networking stuff.
It looks a way better if in the pillar.

    network:
      {%- if salt['grains.get']('os_family') == 'Debian' %}
      pkgs:
        purged:
          - resolvconf
        installed:
          - openvswitch-switch
      {% endif -%}
      
      interfaces:
        - name: eth0
          proto: dhcp
          type: eth
          
Maybe rename `network:pkgs` to `network:packages` ?

**Comment by bechtoldt** [comment]
AFAIR you're accessing master's grains not the minion's grains. https://github.com/bechtoldt/saltstack-tools-formula should be able to manage this.

**Comment by pcdummy** [comment]
Hey @bechtoldt your sure about master's grains?

File: network/interfaces.sls [hunk 2]
```diff
@@ -12,7 +12,17 @@
   {%- endif -%}
 {%- endmacro -%}
 
-{% if salt['pillar.get']('network:interfaces', False) %}
+{%- if 'dhclient_conf' in salt['pillar.get']('network') %}
```
Comment (arnisoph) [inline network/interfaces.sls]: please change to

`{%- if salt['pillar.get']('network:dhclient_conf', False) %}`

for the case the `network:dhclient_conf` pillar is undefined.

File: network/interfaces.sls [hunk 3]
```diff
@@ -12,7 +12,17 @@
...
+dhclient_conf:
+  file.managed:
```
Comment (arnisoph) [inline network/interfaces.sls]: please change to

```
file:
  - managed
```

for the sake of uniformity. :)

File: network/packages.sls [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!jinja|yaml
+
+{% for action, pkgs in salt['pillar.get']('network:pkgs', {}).iteritems() %}
+network_pkgs_{{ action }}:
+  pkg.{{ action }}:
```
Comment (arnisoph) [inline network/packages.sls]: please change to

```
pkg:
  - {{ action }}
```

for the sake of uniformity. :)

**Comment by pcdummy** [comment]
Hey Arnold, do you commit that? :)

**Comment by bechtoldt** [comment]
@pcdummy could you please remove package mgmt from https://github.com/bechtoldt/saltstack-network-formula/blob/master/network/interfaces.sls#L60 to prevent duplicated code?

**Comment by bechtoldt** [comment]
ping @pcdummy :)

**Pull request closed without merging**